src/azul/__init__.py

```diff
@@ -800,6 +801,22 @@ def aggregation_lambda_timeout(self, *, retry: bool) -> int:
     #
     api_gateway_timeout_padding = 2
 
+    # This is set dynamically at runtime
+    lambda_context: Optional[LambdaContext] = None
+
+    @property
+    def terra_client_timeout(self) -> float:
+        value = os.environ['AZUL_TERRA_TIMEOUT']
+        short_timeout, long_timeout = map(float, value.split(':'))
+        require(short_timeout <= long_timeout, short_timeout, long_timeout)
+        if self.lambda_context is None:
+            return long_timeout
+        else:
+            remaining = self.lambda_context.get_remaining_time_in_millis()
+            return (short_timeout
+                    if remaining <= self.api_gateway_timeout + self.api_gateway_timeout_padding
+                    else long_timeout)
```

Comment:
```suggestion
            return (short_timeout
                    if remaining <= self.api_gateway_timeout + self.api_gateway_timeout_padding else
                    long_timeout)
```

If you think about it, this is the most symmetric layout, especially if the two branches are more complex expressions.

Reply:
We should also extract `self.api_gateway_timeout + self.api_gateway_timeout_padding` into a common config property to reduce the brittleness.